### PR TITLE
feat: verbose logging using tracebit

### PR DESCRIPTION
**Pull request by 3Hren**
A new HPACK header is now analyzed: a `trace_bit` of a boolean type. Putting such header triggers verbose mode, when all logs are written with no severity check. This mimics the previous logic, when the same behaviour was achieved just by presence of a `trace_id` header.

### CODE REVIEW

File: src/session.cpp [hunk 1]
```diff
@@ -247,14 +247,22 @@ session_t::handle(const decoder_t::message_type& message) {
                 throw std::system_error(error::revoked_channel, std::to_string(channel_id));
             }
 
-            auto trace_header = hpack::header::find_first<hpack::headers::trace_id<>>(message.headers());
-            auto span_header = hpack::header::find_first<hpack::headers::span_id<>>(message.headers());
-            auto parent_header = hpack::header::find_first<hpack::headers::parent_id<>>(message.headers());
+            auto& headers = message.headers();
```
Comment (antmat) [inline src/session.cpp]: why not const?

File: debian/changelog [hunk 2]
```diff
@@ -1,3 +1,15 @@
+cocaine-core (0.12.12.4) unstable; urgency=low
```
Comment (antmat) [inline debian/changelog]: This will be built automatically. Are you sure?

File: include/cocaine/trace/trace.hpp [hunk 3]
```diff
@@ -84,6 +82,10 @@ class trace_t
     uint64_t
     get_id() const;
 
+    /// Returns true if a tracebit was set, indicating that more verbose output is required.
```
Comment (antmat) [inline include/cocaine/trace/trace.hpp]: Either use /* */ or change to /// in the whole file

File: src/decoder.cpp [hunk 4]
```diff
@@ -52,7 +50,7 @@ decoded_message_t::args() const -> const msgpack::object& {
 }
 
 auto
-decoded_message_t::headers() const -> const hpack::header_storage_t& {
+decoded_message_t::headers() const -> const hpack::headers_t& {
```
Comment (antmat) [inline src/decoder.cpp]: Can we get rid of header_storage_t at all?

File: debian/changelog (hunk 2 above)
Comment (3Hren) [inline debian/changelog]: Yep. I'll be fast enough to discard the build.

File: include/cocaine/trace/trace.hpp (hunk 3 above)
Comment (3Hren) [inline include/cocaine/trace/trace.hpp]: In a separate commit.

File: src/decoder.cpp (hunk 4 above)
Comment (3Hren) [inline src/decoder.cpp]: Yep, in a separate commit.

File: src/session.cpp (hunk 1 above)
Comment (3Hren) [inline src/session.cpp]: This will be automatically deduced as `const T&`.

**Comment by 3Hren** [comment]
Will revert all changes out of the scope and push them separately.

**Comment by 3Hren** [comment]
Pushed without code style changes.

Considering ABI breakage - we can do two things:
 - Treat first/last bit of a trace id as a trace bit - that won't break an ABI, but adds some magic (and requires fix in proxy).
 - Break ABI and bump to 0.12.13.0.
 - We don't give a fuck.

**Pull request merged**